---
title: "25 examples"
site: workflowr::wflow_site
output:
  workflowr::wflow_html:
    toc: true
---
```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, message = FALSE)
```


# Welcome
This is a small list with 25 visualization using gt Package.

## What is this?

GT package is one of the most amazing package to create tables, and we want to show a gallery of examples with full R code to encourage you to use it in your projects.

* How to start with GT Tables
* How to customize a basic table
* Examples about how to use


## How to create a good table?
An informal definition could be: "A good table is used to read a set of numerical data in the quickest and easiest way"


# 25 examples about how to use this amazing package

## 1. Vertical table
```{r vertical_table, message=FALSE, warning=TRUE, paged.print=FALSE}
library(gt)
library(tidyverse)
library(glue)

# Define the start and end dates for the data range
start_date <- "2010-06-07"
end_date <- "2010-06-14"

# Create a gt table based on preprocessed
# `sp500` table data
sp500 %>%
  dplyr::filter(date >= start_date & date <= end_date) %>%
  dplyr::select(-adj_close) %>%
  dplyr::mutate(date = as.character(date)) %>%
  gt() %>%
  tab_header(
    title = "S&P 500",
    subtitle = glue::glue("{start_date} to {end_date}")
  ) %>%
  fmt_date(
    columns = vars(date),
    date_style = 3
  ) %>%
  fmt_currency(
    columns = vars(open, high, low, close),
    currency = "USD"
  ) %>%
  fmt_number(
    columns = vars(volume),
    scale_by = 1 / 1E9,
    pattern = "{x}B"
  )
```

## 2. Horizontal table

## 3. Table with references

This example is from [Table
1](http://www.aclweb.org/anthology/W11-0705)
From proceedings of the Workshop on Language in Social Media (LSM 2011), pages 30â€“38, Portland, Oregon, 23 June 2011. (c) 2011 Association for Computational Linguistics

```{r tab3_reference}
# the table's data
exa <- data.frame( acronym = c("gr8, gr8t", "lol", "rotf", "bff"),
                  english = c("great","laughing out loud", "rolling on the floor", "best friend forever"))

# create the gt table
gt(exa) %>%
  cols_align("left") %>%
  cols_label(acronym="Acronym",
             english="English expansion") %>%
  tab_source_note(
    source_note = "Table 1: Example acrynom and their expansion in the acronym dictionary."
  )

```

## 4. Table with spanning columns

This example is [Table
S2](http://www.genetics.org/content/genetics/suppl/2012/06/28/genetics.112.142448.DC1/TableS2.pdf)
in [Broman et al. (2015) Genetics 192:267-279
doi:10.1534/genetics.112.142448](https://doi.org/10.1534/genetics.112.142448)

```{r tab4_spanners}
# the table's data
tab <- data.frame(n=c(300, 450, 600),
                  all_part_all_crosses = c(4.56, 4.51, 4.49),
                  all_part_min_crosses = c(4.48, 4.47, 4.44),
                  tree_part_all_crosses = c(4.43, 4.36, 4.32),
                  tree_part_min_crosses = c(4.33, 4.33, 4.29))

# create the gt table
gt(tab) %>%
    cols_align("center") %>%
    cols_label(n="total sample size",
               all_part_all_crosses="all crosses",
               all_part_min_crosses="min crosses",
               tree_part_all_crosses="all crosses",
               tree_part_min_crosses="min crosses") %>%
    tab_spanner(label="Tree partitions",
                starts_with("tree")) %>%
    tab_spanner(label="All partitions",
                starts_with("all"))
```


## 5. Counts and percentages

This example is Table S2 of [Lobo et al. bioRxiv
doi:10.1101/529040](https://doi.org/10.1101/529040), with columns
containing both counts and percentages, with the percentages in
parentheses.

```{r tab5_counts_and_percents}
tab <- data.frame(A_count = c(8863572, 2870063, 671722),
                  A_proportion=c(0.853601028762177, 0.727484285290261, 0.556184650236973),
                  B_count = c(1520169, 1075126, 536010),
                  B_proportion=c(0.146398971237823, 0.272515714709739, 0.443815349763027),
                  row.names=c("AA", "AB", "BB"))
tab$genotype <- rownames(tab)

gt(tab, rowname_col="genotype") %>%
    fmt_percent(ends_with("proportion"),
                decimals=1,
                pattern="({x})") %>%
    cols_align("center") %>%
    cols_align("right", columns=ends_with("count")) %>%
    cols_label(A_count = "A count",
               A_proportion = "(%)",
               B_count = "B count",
               B_proportion = "(%)") %>%
    tab_spanner(label="allele in DO-360 microbiome",
                columns=TRUE) %>%
    tab_stubhead_label("DO-360 genotype")
```

## 6. Counts and percentages, with an extra column.

This example is Table S4 of [Lobo et al. bioRxiv
doi:10.1101/529040](https://doi.org/10.1101/529040).
It is much like example 5, but has an additional column at the beginning.

```{r tab6_counts_and_percents}
tab <- data.frame(DO358_genotype = c("AA", "AA", "AA", "AB", "AB", "AB", "BB", "BB", "BB"),
                  DO344_genotype = c("AA", "AB", "BB", "AA", "AB", "BB", "AA", "AB", "BB"),
                  A_count = c(2394215, 869613, 103036, 686970, 297500, 55982, 73727, 47000, 542),
                  A_proportion = c(0.99747944497795, 0.794823306181542, 0.590911176362635,
                                   0.718274560155246, 0.51429835873996, 0.299220173924198,
                                   0.428685226532701, 0.219100940269354, 0.00457986885689177),
                   B_count = c(6050, 224483, 71332, 269447, 280958, 131111, 98257, 167513, 117802),
                   B_proportion = c(0.00252055502204965, 0.205176693818458, 0.409088823637365,
                                    0.281725439844754, 0.48570164126004, 0.700779826075802,
                                    0.571314773467299, 0.780899059730646, 0.995420131143108),
                   stringsAsFactors=FALSE)

gt(tab) %>%
    fmt_percent(ends_with("proportion"),
                decimals=1,
                pattern="({x})") %>%
    cols_align("right") %>%
    cols_align("center", columns=ends_with("genotype")) %>%
    cols_label(DO358_genotype="DO 358 genotype",
               DO344_genotype="DO 344 genotype",
               A_count = "A count",
               A_proportion = "(%)",
               B_count = "B count",
               B_proportion = "(%)") %>%
    tab_spanner(label="allele in DO-358 microbiome",
                columns=matches("^[AB]_"))
```

## 7. Table of coefficients

For a table of estimated regression coefficients from the fit of a
linear model, you might first want to use the
[broom](https://broom.tidyverse.org) package to tidy the output.

```{r tab7_lm_broom}
lm_out <- lm(Volume ~ Height * Girth, data=trees)
broom::tidy(lm_out) %>%
    gt(rowname_col="term") %>%
    tab_stubhead_label("term") %>%
    fmt_number(column=matches("^[^p]"), decimals=2) %>%
    fmt_number(column=matches("^p"), decimals=6) %>%
    cols_label(std.error="SE",
               statistic="t stat")
```


## 8. Compare of fit of multiple model

To compare the fit of multiple models, you may be interested in the
[gtsummary](https://github.com/vincentarelbundock/gtsummary) package.
For example, here is a comparison of the fit of a linear model with and
without an interaction term.

```{r tab8_gt}
lm_add <- lm(Volume ~ Height + Girth, data=trees)
lm_int <- lm(Volume ~ Height * Girth, data=trees)
list(add=lm_add, int=lm_int) %>%
    gtsummary::gtsummary(statistic="conf.int") %>%
    cols_label(add="Additive",
               int="Interactive")
```
